### PR TITLE
Do not restore $hp when returning from a context

### PR DESCRIPTION
**Pull request by Dentosal**
Spec PR https://github.com/FuelLabs/fuel-specs/pull/506

### CODE REVIEW

**Review by vaivaswatha — APPROVED** [review]

**Comment by vaivaswatha** [comment]
Closes https://github.com/FuelLabs/sway/issues/4828

**Review by vaivaswatha — DISMISSED** [review]

**Review by xgreenx — DISMISSED** [review]
Let's wait for @Voxelot's approve=)

**Comment by xgreenx** [comment]
Could you also update CHANELOG.md please?

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,7 +27,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 - [#499](https://github.com/FuelLabs/fuel-vm/pull/499/): The `wasm_bindgen` support of `fuel-asm` and `fuel-types`.
     Each new release also publish a typescript analog of the `fuel-asm` and `fuel-types` crates to the npm.
-    
+### Changed
```
Comment (xgreenx) [inline CHANGELOG.md:30]: A wrong release section=D

Comment (Dentosal) [reply CHANGELOG.md]: Oops. Fixed.

**Review by Voxelot — APPROVED** [review]

**Pull request merged**